contracts_use_cases.html

```diff
@@ -156,8 +156,14 @@ <h2>Use cases &mdash; table</h2>
 
 <tr><td></td>
   <td>C++ Developer</td>
-  <td>Use STL in-contract</td>
-  <td>Codify existing exposition-only STL requirements</td>
+  <td>Use the standard library in-contract</td>
+  <td>Codify existing exposition-only standard library requirements</td>
+  <td></td></tr>
+
+<tr><td><a href="#lib.rtc.preassert">lib.rtc.preassert</a><</td>
+  <td>C++ Developer</td>
+  <td>Assist run-time bug detection</td>
+  <td>Annotate that some assertions should be treated as preconditions</td>
```

Comment:
Why should some assertions being preconditions? Why can't preconditions be preconditions? This comes *before* C++ Developer.

  | Establish a contract | Have validation inform me which output values are unexpected or invalid
  |   | Have validation inform user which input values are unexpected or invalid
  |   | Have contracts specify their pre-conditions as logical predicates

Reply:
Note I'm not against it. Nothing wrong with asking this question.

Reply:
The extended description does clarify this, but I do think the short description needs to capture it better.  How about "Annotate that some assertions in the function body are the function caller's responsibility to satisfy"

Reply:
This sounds good

Reply:
My bad for not reading the extended description -- but why would "precondition-like" assertions be cheaper than preconditions? It seems this assumption is at the heart of the use case.

Otherwise we still have:

```API Developer  | Establish responsibility boundaries | Inform users which errors are the responsibility of the caller, and which are the callee```

Reply:
Example: function `binary_search` (complexity O(log *n*) has precondition `is_sorted` (complexity O(*n*)). So, I will be checking this only in audit builds or maybe ever. But on the other hand, inside function body, each time I am doing one of these log *n* comparisons and jumps, I can make log *n* sanity checks, which does not increase the complexity.

Reply:
I would like

```As an API Developer, In Order to Establish responsibility boundaries, Annotate that assertions that indirectly test preconditions (such as malformed data discovered while performing the algorithm) should be reported to the caller as precondition failures```

Reply:
That sounds good to me.  I'll push in a change with that (and a few other minor typos I found locally on my work computer) later tonight or first thing tomorrow morning.